File: Semester3/Advanced Programming Methods/Labs/lab3/src/domain/validators/StudentValidator.java
package domain.validators;

import domain.entities.Student;

import java.util.function.Predicate;

public class StudentValidator implements Validator<Student> {
    /**
     * Overriding validate method for student entites
     * @param entity - Student to be validated
     * @throws ValidatorException - if the student is invalid
     */
    @Override
    public void validate(Student entity) throws ValidatorException, IllegalArgumentException {
        if(entity == null){
            throw new IllegalArgumentException("Studentul nu poate fi null!");
        }
        String err = "";

        Predicate<Student> negativeId = x -> x.getID() < 0;
        Predicate<Student> emptyEmail = x -> x.getEmail().equals("");
        Predicate<Student> negativeGroup = x -> x.getGroup() <= 0;
        Predicate<Student> emptyName = x -> x.getName().equals("");
        Predicate<String> emptyErrs = x -> x.equals("");

        if(negativeGroup.test(entity)){
            err += "Id must be a positive integer!\n";
        }
        if(emptyEmail.test(entity)) {
            err += "Email can't be empty!\n";
        }
        if(negativeGroup.test(entity)){
            err += "Group must be a positive integer!";
        }
        if(emptyName.test(entity)){
            err += "Name can't be empty!\n";
        }

        if(!emptyErrs.test(err)){
            throw new ValidatorException(err);
        }
    }
}

File: Semester3/Advanced Programming Methods/Labs/StudentsManagementAppC#/StudentsManagementApp/domain/validator/StudentValidator.cs
using StudentsManagementApp.domain.entities;

namespace StudentsManagementApp.domain.validator
{
	public class StudentValidator : IValidator<Student>
	{
		/// <inheritdoc />
		/// <summary>
		/// Function to validate a student
		/// </summary>
		/// <param name="entity"> Student to be validated </param>
		/// <exception cref="ValidatorException">throws ValidatorException if Student fields are incorrect</exception>
		public void Validate(Student entity)
		{
			var errs = "";
			if (entity.Id < 0)
			{
				errs += "Id must be a positive integer!\n";
			}

			if (entity.Email.Equals(""))
			{
				errs += "Email can't be empty!\n";
			}

			if (entity.Group < 0)
			{
				errs += "Group must be a positive integer!\n";
			}

			if (entity.Name.Equals(""))
			{
				errs += "Name can't be empty!\n";
			}

			if (!string.IsNullOrEmpty(errs))
			{
				throw  new ValidatorException(errs);
			}
		}
	}
}